src/ota.rs

```diff
@@ -152,9 +152,14 @@ impl<'a> EspOtaUpdate<'a> {
         self.check_write()?;
 
         esp!(unsafe { esp_ota_end(self.update_handle) })?;
+        let update_partition = self.update_partition;
+
+        // `Drop::drop` must not be called on `EspOtaUpdate` after the OTA handle has been
+        // invalidated.
+        std::mem::forget(self);
```

Comment:
Only pls change to `core::mem::forget`, or else the no_std build would fail. Don't ask why we have no_std build... a bit of a hygiene (to make sure we are not overusing std apis where not really necessary), not much else.